### PR TITLE
better neural fail detection in benchmark figures

### PR DESCRIPTION
**Pull request by grfrederic**
neural_fail = neural_estimator AND (estimate is negative OR estimate < 0.1 * max estimate over all seeds)

seems like this works well for our data.

### CODE REVIEW

File: workflows/_common_figure_utils.py [hunk 1]
```diff
@@ -144,8 +144,13 @@ def preprocess_benchmark_results(results, estimators=ESTIMATORS):
     data["neural_estimator"] = data["estimator_id"].isin(neural_ids)
 
     # detect neural convergence issues
-    # TODO(frdrc): we can do better than a flat 0.1
-    data["neural_fail"] = (data["mi_estimate"] < 0.1) & data["neural_estimator"]
+    data_idx = data.set_index(["estimator_id", "task_id", "n_samples", "seed"])
+    data_idx["mi_estimate_max"] = data_idx.groupby(level=[0, 1, 2])["mi_estimate"].max()
```
Comment (pawel-czyz) [inline workflows/_common_figure_utils.py:148]: Is it possible that all `"mi_estimate"` are `nan` and `.max()` would raise?

**Review by pawel-czyz — APPROVED** [review]
I like it!

File: workflows/_common_figure_utils.py (hunk 1 above)
Comment (grfrederic) [reply workflows/_common_figure_utils.py:148]: pandas returns nan then ;)

Comment (pawel-czyz) [reply workflows/_common_figure_utils.py:148]: Nice! Btw, have you seen https://github.com/gventuri/pandas-ai ?

Comment (grfrederic) [reply workflows/_common_figure_utils.py:148]: woo thats very cool

Comment (pawel-czyz) [reply workflows/_common_figure_utils.py:148]: Feel free to resolve and merge :slightly_smiling_face:

**Pull request merged**